library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphTest.java

```diff
@@ -118,35 +185,121 @@ public void shouldThrowUnsupportedExceptionForComputeWithClass() {
     public void shouldThrowUnsupportedExceptionForTx() {
         // Given
         final Graph gafferGraph = getGafferGraph();
-        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION, gafferGraph);
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
 
         // When / Then
         assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(() -> graph.tx());
     }
 
+    @Test
+    public void shouldThrowIllegalArgumentExceptionForNoVertexLabel() {
+        // Given
+        final Graph gafferGraph = getGafferGraph();
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
+
+        //Then
+        assertThatExceptionOfType(IllegalArgumentException.class)
+            .isThrownBy(() -> graph.addVertex(T.id, VERTEX_1))
+            .withMessageMatching("Label is required");
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionForNoVertexId() {
+        // Given
+        final Graph gafferGraph = getGafferGraph();
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
+
+        //Then
+        assertThatExceptionOfType(IllegalArgumentException.class)
+            .isThrownBy(() -> graph.addVertex(T.label, SOFTWARE_NAME_GROUP))
+            .withMessageMatching("ID is required");
+    }
+
     @Test
     public void shouldAddAndGetVertex() {
         // Given
         final Graph gafferGraph = getGafferGraph();
-        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION, gafferGraph);
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
 
         // When
         graph.addVertex(T.label, SOFTWARE_NAME_GROUP, T.id, VERTEX_1, NAME_PROPERTY, "GafferPop");
         final Iterator<GafferPopVertex> vertices = graph.vertices(Arrays.asList(VERTEX_1, VERTEX_2), SOFTWARE_NAME_GROUP);
 
+        // Then
+        final GafferPopVertex vertex = vertices.next();
+        assertFalse(vertices.hasNext());
```

Comment:
@GCHQDeveloper314 what about this?
```suggestion
        assertThat(vertices).isExhausted();
```